src/EFCore.PG.NTS/Query/ExpressionTranslators/Internal/NpgsqlNetTopologySuiteMemberTranslatorPlugin.cs

```diff
@@ -86,8 +86,33 @@ public Expression Translate(MemberExpression e)
                     return new SqlFunctionExpression("ST_NumInteriorRings", typeof(int), new[] { e.Expression });
                 case "NumPoints":
                     return new SqlFunctionExpression("ST_NumPoints", typeof(int),       new[] { e.Expression });
+                case "OgcGeometryType":
+                {
+                    var whenThenList = new List<CaseWhenClause>
+                    {
+                        new CaseWhenClause(Expression.Constant("ST_CircularString"), Expression.Constant(OgcGeometryType.CircularString)),
+                        new CaseWhenClause(Expression.Constant("ST_CompoundCurve"), Expression.Constant(OgcGeometryType.CompoundCurve)),
+                        new CaseWhenClause(Expression.Constant("ST_CurvePolygon"), Expression.Constant(OgcGeometryType.CurvePolygon)),
+                        new CaseWhenClause(Expression.Constant("ST_GeometryCollection"), Expression.Constant(OgcGeometryType.GeometryCollection)),
+                        new CaseWhenClause(Expression.Constant("ST_LineString"), Expression.Constant(OgcGeometryType.LineString)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiCurve"), Expression.Constant(OgcGeometryType.MultiCurve)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiLineString"), Expression.Constant(OgcGeometryType.MultiLineString)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiPoint"), Expression.Constant(OgcGeometryType.MultiPoint)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiPolygon"), Expression.Constant(OgcGeometryType.MultiPolygon)),
+                        new CaseWhenClause(Expression.Constant("ST_MultiSurface"), Expression.Constant(OgcGeometryType.MultiSurface)),
+                        new CaseWhenClause(Expression.Constant("ST_Point"), Expression.Constant(OgcGeometryType.Point)),
+                        new CaseWhenClause(Expression.Constant("ST_Polygon"), Expression.Constant(OgcGeometryType.Polygon)),
+                        new CaseWhenClause(Expression.Constant("ST_PolyhedralSurface"), Expression.Constant(OgcGeometryType.PolyhedralSurface)),
+                        new CaseWhenClause(Expression.Constant("ST_Tin"), Expression.Constant(OgcGeometryType.TIN))
+                    };
+
+                    return new CaseExpression(
+                        new SqlFunctionExpression("ST_GeometryType", typeof(string), new[] { e.Expression }),
+                        whenThenList.ToArray());
```

Comment:
And not to produce an array here from the list. Otherwise, there are two meaningless memory allocations. By the way, this collection can be made static because it doesn't access any locals.

Reply:
Moved it to a static array, should cover both comments